hoot-core/src/main/cpp/hoot/core/algorithms/linearreference/WaySubline.cpp

```diff
@@ -74,8 +74,11 @@ WaySubline::WaySubline(const WayLocation& start, const WayLocation& end) :
 
 WaySubline& WaySubline::operator=(const WaySubline& from)
 {
-  _start = from.getStart();
-  _end = from.getEnd();
+  if (this != &from)
+  {
+    _start = from.getStart();
+    _end = from.getEnd();
+  }
```

Comment:
Best practice is to do nothing if you come across something like this `x = x;`  In some cases this can cause memory to be cleaned up and then reused.  None of these changes exhibited that error but it is good practice.